sources/packages/api/src/services/institution-location/institution-location.service.ts

```diff
@@ -217,4 +219,35 @@ export class InstitutionLocationService extends RecordDataModelService<Instituti
       .where("location.id = :locationId", { locationId })
       .getRawOne();
   }
+
+  /**
+   * Get an institution's locations with designation statuses.
+   * @param institutionId institution id
+   * @returns Institution Locations with designation statuses.
+   */
+  async getInstitutionLocationsWithDesignationStatus(
+    institutionId: number,
+  ): Promise<LocationWithDesignationStatus[]> {
+    return this.repo
+      .createQueryBuilder("location")
+      .addSelect("location.name", "locationName")
+      .addSelect("location.info", "locationAddress")
+      .addSelect("location.id", "id")
+      .addSelect("location.institution_code", "institutionCode")
```

Comment:
Please avoid using the DB column names. We should use the properties mapped on the entities models. Same for `primary_contact`.